turbo/engineapi/engine_block_downloader/core.go

```diff
@@ -32,7 +32,7 @@ import (
 func (e *EngineBlockDownloader) download(ctx context.Context, hashToDownload common.Hash, requestId int, block *types.Block) {
 	/* Start download process*/
 	// First we schedule the headers download process
-	if !e.scheduleHeadersDownload(requestId, hashToDownload, 0) {
+	if !e.scheduleHeadersDownload(requestId, hashToDownload, block.NumberU64()) {
```

Comment:
This check doesn't look right. `block` in `download` is different from the block of `hashToDownload` - see the comment before `StartDownloading`. Perhaps we should also rename `block` to `blockTip` here and copy the comment.

Reply:
I changed `hashToDownload` to be the same as the hash of the block tip.